src/game.cpp

```diff
@@ -13936,6 +13936,30 @@ overmap &game::get_cur_om() const
     return overmap_buffer.get( pos_om.x, pos_om.y );
 }
 
+std::string game::get_cur_om_id() const
+{
+    // The player is located in the middle submap of the map.
+    const tripoint sm = m.get_abs_sub() + tripoint( MAPSIZE / 2, MAPSIZE / 2, 0 );
+    const tripoint pos_om = sm_to_om_copy( sm );
+    const oter_id &ter = overmap_buffer.ter( pos_om );
+    const std::string om_id = ter.id().c_str();
```

Comment:
Why this variable (why not return it directly), and why initialize it via a C-string? (`string_id` has a `str` function that returns a `std::string`.)